crates/bevy_render/src/mesh/shape/cylinder.rs

```diff
@@ -0,0 +1,125 @@
+use crate::mesh::{Indices, Mesh};
+use wgpu::PrimitiveTopology;
+
+/// A cylinder which stands on the XZ plane
+pub struct Cylinder {
+    /// Radius in the XZ plane.
+    pub radius: f32,
+    /// Height of the cylinder in the Y axis.
+    pub height: f32,
+    /// The number of vertices around each horizontal slice of the cylinder. If you are looking at the cylinder from
+    /// above, this is the number of points you will see on the circle.
+    /// A higher number will make it appear more circular.
+    pub resolution: u32,
+    /// The number of segments between the two ends.
+    pub segments: u32,
+}
+
+impl Default for Cylinder {
+    fn default() -> Self {
+        Self {
+            radius: 0.5,
+            height: 1.0,
+            resolution: 16,
+            segments: 4,
```

Comment:
Does anyone know if I can set the number of segments to 1? It looks the same. I'm not sure what the advantage of it being 4 is.

Reply:
If I understand correctly, I think the number of segments determines the number of horizontal slices between the top and bottom of the cylinder? So 1 would have triangles spanning the full height of the cylinder (2 slices, bottom and top), while 2 would have 2 sets of triangles (3 slices, bottom, middle, top), etc. The segment docs could use this explanation.

Unless anyone else knows otherwise, I would think using 1 segment is fine. 

I would use the wireframe plugin, and check our assumptions on how the mesh looks when changing the different parameters.

Reply:
I checked the wireframe. It works as we expected. With segments set to 1,  the triangles go from top to bottom. Setting it to 2 has two sets of triangles with a horizontal slice in the middle. I added this info to the docs on `segments`. And I updated the default number of segments to 1.